woocommerce/woocommerce-paypal-payments
# Fix Error: Detected container element removed from DOM

**Issue**: [PCP-93](https://inpsyde.atlassian.net/browse/PCP-93).
---

### Description

Fix error in the browser console because of PayPal button rendering was called twice, the first time was too early.

Don't render PayPal button immediately when initiating checkout. Instead, call `render` on `updated_checkout` event only. This event always triggered by WooCommerce after the checkout is loaded. So, this change prevents PayPal buttons from being rendered twice.

See https://github.com/woocommerce/woocommerce/blob/master/assets/js/frontend/checkout.js for more details.

### Steps to test:


0. Build assets (`npm run build:dev` should be enough) and clean your browser cache to be sure you have the latest ones.
1. Add any product to the cart.
2. Open browser console.
3. Go to the checkout page.
4. Refresh the page.

You shouldn't see the error message 'Detected container element removed from DOM' in the console if everything is ok.

### Documentation




### Changelog entry
> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
* Fix - Error message 'Detected container element removed from DOM' on checkout in the browser console.
